### PR TITLE
fix: cosign image pulls

### LINKED ISSUE #2591: Package create breaks when a cosign signature image exists in the package and in the cache
### Environment
Device and OS: Ubuntu Linux (but pretty sure will happen with any OS)
App version: v0.34.0

### Steps to reproduce
This will occur with any package that uses a cosign image such as `ghcr.io/stefanprodan/podinfo:sha256-57a654ace69ec02ba8973093b6a786faa15640575fbf0dbb603db55aca2ccec8.sig`
1. zarf package create examples/helm-charts
2. zarf package create examples/helm-charts again

### Expected result
The package build will work fine

### Actual Result
The package build fails with the following error
image

### Severity/Priority
Medium, while we have not yet heard of someone running into this we should fix this before the next release

By running `zarf tools clear-cache` this resolve the error (until the next run)

### PR DESCRIPTION
**Pull request by AustinAbro321**
Fixes #2591 

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/.github/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -5,7 +5,6 @@ charset = utf-8
 end_of_line = lf
 indent_size = 2
 indent_style = space
-insert_final_newline = true
```
Comment (AustinAbro321) [inline .editorconfig:8]: This was getting annoying when working with sha files, should be covered by pre-commit

File: src/internal/packager/images/pull_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+func TestPull(t *testing.T) {
+	t.Run("pulling a cosign image is successful and doesn't add anything to the cache", func(t *testing.T) {
+
+		ref, err := transform.ParseImageRef("ghcr.io/stefanprodan/podinfo:sha256-57a654ace69ec02ba8973093b6a786faa15640575fbf0dbb603db55aca2ccec8.sig")
+		require.NoError(t, err)
+		tmpDestDir := t.TempDir()
+		tmpCacheDir := t.TempDir()
+		destDir := filepath.Join(tmpDestDir, "images")
+		PullConfig := PullConfig{
```
Comment (lucasrod16) [inline src/internal/packager/images/pull_test.go:25]: `PullConfig` is a local, unexported variable, so should start with a lowercase character

File: src/internal/packager/images/pull_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			CacheDirectory:       tmpCacheDir,
+			ImageList: []transform.Image{
+				ref,
+			},
+		}
+
+		_, err = Pull(context.Background(), PullConfig)
+		require.NoError(t, err)
+		_, err = os.Stat(filepath.Join(destDir, "blobs/sha256/3e84ea487b4c52a3299cf2996f70e7e1721236a0998da33a0e30107108486b3e"))
+		require.NoError(t, err)
```
Comment (lucasrod16) [inline src/internal/packager/images/pull_test.go:36]: We should use `require.FileExists` here

**Review by lucasrod16 — CHANGES_REQUESTED** [review]
Few small changes, otherwise lgtm

File: src/internal/packager/images/pull_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+			ImageList: []transform.Image{
+				ref,
+			},
+		}
+
+		_, err = Pull(context.Background(), pullConfig)
+		require.NoError(t, err)
+		require.FileExists(t, filepath.Join(destDir, "blobs/sha256/3e84ea487b4c52a3299cf2996f70e7e1721236a0998da33a0e30107108486b3e"))
+
+		files, err := filepath.Glob(filepath.Join(tmpCacheDir, "*"))
```
[the remark this answers is not in the document]
Comment (AustinAbro321) [reply src/internal/packager/images/pull_test.go]: Ah yeah, was just looking for a way to check if a dir is empty, but yeah agree readdir is simpler.

**Review by lucasrod16 — APPROVED** [review]

**Review by lucasrod16 — APPROVED** [review]

**Pull request merged**